### PR TITLE
Make Vale upgradable by Dependabot

### PR DESCRIPTION
**Pull request by ddbeck**
This PR does a few things with the aim of making Vale eligible for upgrading by Dependabot:

- It declares the Vale version via a Dockerfile, so that Dependabot can spot when there's a new version of Vale.
- It uses Docker Compose to run Vale, so that the Buildkite pipeline can use the custom Dockerfile (if this feels a bit roundabout, it's because the Buildkite docker plugin doesn't handle building images)
- It creates a shell script, `vale.sh`, to encapsulate the args for `vale` itself. This has a side-effect of making it more convenient to run Vale locally.

### CODE REVIEW

**Review by plaindocs — APPROVED** [review]
This is neat!

`vale.sh` should probably live in scripts though.

And part of me wants to "tidy up" all the Docker stuff as well, but I don't think that is necessary.

**Comment by ddbeck** [comment]
> And part of me wants to "tidy up" all the Docker stuff as well, but I don't think that is necessary.

So I almost did that in my original, but it made the diff hard to follow. But now that you've seen this version, I'll tidy up.

**Comment by ddbeck** [comment]
## Before reorganization

```
.
├── Dockerfile.vale
├── docker-compose-lint.yml
├── docker-compose.production.yml
├── vale
│   ├── Buildkite
│   │   ├── eg.yml
│   │   ├── etc.yml
│   │   ├── existence.yml
│   │   ├── git-merge.yml
│   │   ├── ie.yml
│   │   ├── spelling.yml
│   │   ├── toc.yml
│   │   └── via.yml
│   ├── Snippets
│   │   └── headings.yml
│   └── vocab.txt
└── vale.sh
```

## After

```
.
├── scripts
│   └── vale.sh
└── vale
    ├── Dockerfile
    ├── docker-compose.yml
    └── styles
        ├── Buildkite
        │   ├── eg.yml
        │   ├── etc.yml
        │   ├── existence.yml
        │   ├── git-merge.yml
        │   ├── ie.yml
        │   ├── spelling.yml
        │   ├── toc.yml
        │   └── via.yml
        ├── Snippets
        │   └── headings.yml
        └── vocab.txt
```

**Comment by plaindocs** [comment]
Nice and tidy!
